src/background/services/events.ts

```diff
@@ -1,8 +1,12 @@
 import { EventEmitter } from 'events'
+import type { Storage } from '@/shared/types'
 
 interface BackgroundEvents {
   'storage.rate_of_pay_update': { rate: string }
-  'storage.host_permissions_update': { status: boolean }
+  'storage.state_update': {
+    state: Storage['state']
+    prevState: Storage['state']
+  }
```

Comment:
Why are we passing the current state and the previous state?

Reply:
Helpful with logging/metrics to know if users get into multiple errors states one after other. Also, we can add transitions between states in popup/icons.